### PR TITLE
Igor/fixes

### PR DESCRIPTION
**Pull request by igor47**
a few related fixes around upload validation, specifically targeting #337 and #337

### CODE REVIEW

**Review by ajhyndman — CHANGES_REQUESTED** [review]
Thanks for prioritizing this!  I have a few comments, mostly minor.

File: src/server/services/validate-upload.js [hunk 1]
```diff
@@ -261,6 +310,10 @@ async function validateUpload (upload, user, trns) {
     validateRules
   ]
 
+  // we should do this in a transaction, unless someone is doing it for us
+  const ourTransaction = !trns
+  if (ourTransaction) trns = await knex.transaction()
```
[the remark this answers is not in the document]
Comment (igor47) [reply src/server/services/validate-upload.js:315]: it's called from the revalidation service, which has it's own semantics about when to commit the transaction

File: src/server/services/validate-upload.js [hunk 2]
```diff
@@ -91,42 +92,90 @@ async function validateReportingPeriod ({ upload, records, trns }) {
...
+      'At least one of UEI or TIN/EIN must be set, but both are missing',
       { col: 'C, D', severity: 'err' }
     ))
   }
 
+  // does the row already exist?
+  let byUei = null
+  if (recipient.Unique_Entity_Identifier__c) {
+    byUei = await findRecipient(upload.tenant_id, recipient.Unique_Entity_Identifier__c, null, trns)
+    console.dir(byUei)
```
[the remark this answers is not in the document]
Comment (igor47) [reply src/server/services/validate-upload.js:107]: oops i need to drop these, thanks

File: src/server/services/validate-upload.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (ajhyndman) [reply src/server/services/validate-upload.js:310]: Ahh.. Okay, fair!

**Review by ajhyndman — APPROVED** [review]
Thanks!  LGTM

**Pull request merged**